src/main/java/org/rulez/demokracia/pdengine/Vote.java

```diff
@@ -85,21 +85,37 @@ public JSONObject toJson(String voteId) {
 
 	public JSONArray createChoicesJson(Map<String, Choice> choices) {
 		JSONArray array = new JSONArray();
-		
+
 		for (Entry<String, Choice> entry : choices.entrySet()) {
-		    String key = entry.getKey();
-		    Choice value = entry.getValue();
-		    
-		    JSONObject obj = new JSONObject();
-		    obj.put("initiator", value.userName);
+			String key = entry.getKey();
+			Choice value = entry.getValue();
+
+			JSONObject obj = new JSONObject();
+			obj.put("initiator", value.userName);
 			obj.put("endorsers", value.endorsers);
 			obj.put("name", value.name);
 			obj.put("id", key);
-		    
-		    array.put(obj);
+
+			array.put(obj);
 		}
-		
+
 		return array;
 	}
 
+	public void addCastVote(String proxyId, List<RankedChoice> theVote, String secretId) {
+		boolean isModified = false;
```

Comment:
Well, I checked that Zombies and as I understand the problem is with the isModified. However I cannot call the for cycle with inital isModified = true.
What should I do? Refactor the code?

Reply:
@magwas

Reply:
If you write test for each of the cases, then the zombie will die.
My bet is that you need to have a testcase where votesCast has multiple elements, and the one with the searched proxyId is not the last.
I did not think through every cases though. If you do that, you will find the zombie.
When you stick to the rule that production code should be modified only with the bare minimum which is needed to make the tests green, then you will not introduce any zombie.
I recommend reading this blog post to understand this better: http://blog.cleancoder.com/uncle-bob/2013/05/27/TheTransformationPriorityPremise.html
Uncle Bob also have a video on it, which I cannot find with a quick search. It is very entertaining, as all of his videos.

Reply:
found it: https://www.youtube.com/watch?v=B93QezwTQpI
I wholeheartedly recommend to take an our watching it. It worth your time.

Reply:
Thank you so much I will check that video. :)